dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datasetutility/ReplaceFileHandlerIT.java

```diff
@@ -53,6 +101,138 @@ public void shouldCreateDataFile() {
+        fmd.setLabel(filename);
+
+        FileTermsOfUse termsOfUse = new FileTermsOfUse();
+        termsOfUse.setLicense(licenseDAO.findFirstActive());
+        fmd.setTermsOfUse(termsOfUse);
+        termsOfUse.setFileMetadata(fmd);
+
+        fmd.setTermsOfUseForm(termsOfUseFormMapper.mapToForm(termsOfUse));
+
+        if(addToDataset) {
+            savedFile.setOwner(datasetVersion.getDataset());
+        }
+
+        fmd.setDataFile(savedFile);
+        savedFile.getFileMetadatas().add(fmd);
+
+        if(addToDataset) {
+            if (datasetVersion.getFileMetadatas() == null) {
+                datasetVersion.setFileMetadatas(new ArrayList<>());
+            }
+            datasetVersion.addFileMetadata(fmd);
+            fmd.setDatasetVersion(datasetVersion);
+            datasetVersion.getDataset().getFiles().add(savedFile);
+        }
+        em.persist(fmd);
+
+        datafile.generateStorageIdentifier(savedFile);
+        savedFile.setChecksumType(checksumType);
+        savedFile.setChecksumValue(filename);
+
+        return savedFile;
+    }
+
+    private Dataverse fillDataverseWithRequiredData(Dataverse dataverse) {
+        dataverse.setName("TestDataverse");
+        dataverse.setAlias("TestDataverseAlias");
+        dataverse.setOwner(dataverseServiceBean.findRootDataverse());
+        dataverse.setDataverseType(Dataverse.DataverseType.LABORATORY);
+        dataverse.setCreateDate(new Timestamp(System.currentTimeMillis()));
+        dataverse.setModificationTime(new Timestamp(System.currentTimeMillis()));
+
+        DataverseContact dataverseContact = new DataverseContact();
+        dataverseContact.setContactEmail("testmail@test.com");
+        List<DataverseContact> dataverseContacts = new ArrayList<>();
+        dataverseContacts.add(dataverseContact);
+        dataverse.setDataverseContacts(dataverseContacts);
+
+        return dataverse;
+    }
+
+    private Dataset fillDatasetWithExtendedData(Dataset dataset) {
+        dataset.setCreateDate(Timestamp.valueOf(LocalDateTime.of
+                (LocalDate.of(2019, 12, 12), LocalTime.of(13, 15))));
+
+        dataset.setModificationTime(Timestamp.valueOf(LocalDateTime.of
+                (LocalDate.of(2019,12,12), LocalTime.of(13,15))));
+
+        dataset.setAuthority("10.5072");
+        dataset.setIdentifier("FK2/AAAAAA");
+        dataset.setProtocol("doi");
+
+        DatasetField datasetField = new DatasetField();
+        datasetField.setSingleValue("testvalue");
+        DatasetFieldType datasetFieldType = new DatasetFieldType("test", FieldType.TEXT, false);
+        datasetFieldType.setMetadataBlock(metadataBlockDao.findById(1L));
+        datasetFieldType.setRequired(false);
+        List<DatasetField> datasetFieldList = new ArrayList<>();
+        datasetField.setDatasetFieldType(datasetFieldType);
+        datasetFieldList.add(datasetField);
+        em.persist(datasetFieldType);
+
+        dataset.getEditVersion().setDatasetFields(datasetFieldList);
```

Comment:
Without setting field list test runs fine

Reply:
We can go further with this: no need for datasetFieldList -> no need to even create DatasetField in this method - it is empty statement (doesn't add value for test)